build.rs

```diff
@@ -538,11 +543,22 @@ fn obj_path(out_dir: &Path, src: &Path, obj_ext: &str) -> PathBuf {
     out_path
 }
 
+// This option will strip the build path from C and assembly files compiled by gcc and clang.
+fn remap_debug_prefix() -> String {
+    use std::env;
+    let cwd_path = env::current_dir().expect("Couldn't get current directory");
+    let mut flag = String::from("-fdebug-prefix-map=");
+    flag.push_str(cwd_path.to_str().expect("Invalid path"));
+    flag.push_str("=.");
+    flag
+}
+
 fn cc(
     file: &Path,
     ext: &str,
     target: &Target,
     warnings_are_errors: bool,
+    extra_flags: Vec<String>,
```

Comment:
I understand you did this at my request so that we wouldn't duplicate the work in `remap_debug_prefix()` per source file. I realize that this complication is really caused by the structure of the build.rs code. I have refactored it in PR #1697 so that changes like the one in this PR don't have to try to solve the issues themselves.